src/routes/recommendationRoutes.js

```diff
@@ -0,0 +1,20 @@
+// recommendationRoutes.js
+const express = require('express');
+const router = express.Router();
+const recommendationController = require('../controllers/recommendationController');
+
+router.get('/:userId', async (req, res) => {
+  const { userId } = req.params;
+  try {
+    const recommendations =
+      await recommendationController.generateAlbumRecommendations(userId);
+    res.json(recommendations);
+  } catch (error) {
+    console.error('Error while generating recommendations:', error);
+    res
+      .status(500)
+      .json({ error: 'An error occurred while generating recommendations.' });
+  }
+});
```

Comment:
Preferably, to follow our current conventions, the code starting with `async (req, res) => ...` should all live in the controller. The router file should just be the definition of "this route is handled by this controller function". So  something like: `router.get('/:userId', getAlbumRecsForUser)`

Reply:
I moved getAlbumRecsForUser function in controller ,commit c89f45fe910de2dad5613d88d052837c137e4b30